### PR TITLE
[#PAB-226] Create SQLAlchemy model for Project

### PR DESCRIPTION
**Pull request by nsnape**
https://trello.com/c/IiJFjd9v

### Change description
New model for Project entity

- [x] Unit tests and other appropriate tests added or updated
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### CODE REVIEW

File: core/db/entities.py [hunk 1]
```diff
@@ -62,42 +51,21 @@ class Package(db.Model):
...
-    )
-    m_and_e_contact_id = sqla.Column(
-        sqla.String,
-        sqla.ForeignKey("contact_dim.id"),
-        nullable=False,
-    )
+    name_contact_id = sqla.Column(sqla.String, sqla.ForeignKey("contact_dim.id"), nullable=False)
+    project_sro_contact_id = sqla.Column(sqla.String, sqla.ForeignKey("contact_dim.id"), nullable=False)
+    cfo_contact_id = sqla.Column(sqla.String, sqla.ForeignKey("contact_dim.id"), nullable=False)
+    m_and_e_contact_id = sqla.Column(sqla.String, sqla.ForeignKey("contact_dim.id"), nullable=False)
```
Comment (cyrusdobbs) [inline core/db/entities.py:68]: could also do this tidying up for the ProjectProgress class for consistency? I just copied and pasted but I think it looks better when it's all on one line like this now that we have the longer line length

File: core/db/entities.py [hunk 2]
```diff
@@ -108,9 +76,34 @@ class Package(db.Model):
...
+
+
+class Project(db.Model):
+    """Stores Project Entities."""
+
+    __tablename__ = "project_dim"
+
+    id = sqla.Column(GUID(), default=uuid.uuid4, primary_key=True)  # this should be UUIDType once using Postgres
+    project_id = sqla.Column(sqla.String, nullable=False, unique=True)
+    # TODO: should this be unique?
```
Comment (cyrusdobbs) [inline core/db/entities.py:94]: Might be less messy if we try to put shorter comments on the same line as the variable?

Comment (cyrusdobbs) [reply core/db/entities.py:94]: Theres a few others like this too

**Review by cyrusdobbs — CHANGES_REQUESTED** [review]

File: core/db/entities.py (hunk 1 above)
Comment (nsnape) [reply core/db/entities.py:68]: yeah, I can do that in this PR

File: core/db/entities.py (hunk 2 above)
Comment (nsnape) [reply core/db/entities.py:94]: I don't mind where they go as the aim is to "do" them ie remove them ASAP. These read better on my screen TBH, so will leave as they are (I'll add a blank line above each to make it clearer). Also, the Git-diff looks cleaner when you remove them if they are not in-line

Comment (cyrusdobbs) [reply core/db/entities.py:95]: makes sense

**Review by cyrusdobbs — APPROVED** [review]

**Pull request merged**